src/Twig/EventRuntime.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace AppBundle\Twig;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Event;
+use AppBundle\Repository\EventRegistrationRepository;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class EventRuntime
+{
+    private $eventRegistrationRepository;
+
+    public function __construct(EventRegistrationRepository $eventRegistrationRepository)
+    {
+        $this->eventRegistrationRepository = $eventRegistrationRepository;
+    }
+
+    public function isAlreadyParticipating(Event $event, ?UserInterface $user): bool
+    {
+        if (!$user instanceof Adherent || !$user) {
```

Comment:
the check `!$user instanceof Adherent` is enough here, no?

Reply:
Yes because $user can be null (unauthenticated user)

Reply:
@tony-tran so if yes, and the first check is enough you will remove the second one?